x-pack/plugins/cases/public/components/cases_context/state/use_is_add_to_case_open.test.tsx

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act, renderHook } from '@testing-library/react-hooks';
+import type { FC, PropsWithChildren } from 'react';
+import React from 'react';
+import { useKibana } from '../../../common/lib/kibana';
+import CasesProvider from '..';
+import { useCasesAddToExistingCaseModal } from '../../all_cases/selector_modal/use_cases_add_to_existing_case_modal';
+import { allCasesPermissions } from '../../../common/mock';
+import { ExternalReferenceAttachmentTypeRegistry } from '../../../client/attachment_framework/external_reference_registry';
+import { PersistableStateAttachmentTypeRegistry } from '../../../client/attachment_framework/persistable_state_registry';
+import { useIsAddToCaseOpen } from './use_is_add_to_case_open';
+import { useApplication } from '../../../common/lib/kibana/use_application';
+import { of } from 'rxjs';
+import { useCasesToast } from '../../../common/use_cases_toast';
+import type { PublicAppInfo } from '@kbn/core/public';
+
+jest.mock('../../../common/use_cases_toast');
+jest.mock('../../../common/lib/kibana');
+jest.mock('../../../common/lib/kibana/use_application');
+jest.mock('../../../common/use_cases_toast');
+jest.mock('../../all_cases/selector_modal/all_cases_selector_modal', () => {
+  return {
+    AllCasesSelectorModal: jest.fn(),
+  };
+});
+
+const externalReferenceAttachmentTypeRegistry = new ExternalReferenceAttachmentTypeRegistry();
+const persistableStateAttachmentTypeRegistry = new PersistableStateAttachmentTypeRegistry();
+const useKibanaMock = useKibana as jest.Mocked<typeof useKibana>;
+const useApplicationMock = useApplication as jest.Mock;
+const useCasesToastMock = useCasesToast as jest.Mock;
+const mockedToastInfo = jest.fn();
+
+describe('use is add to existing case modal open hook', () => {
+  let wrapper: FC<PropsWithChildren<unknown>>;
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    useKibanaMock().services.application = {
+      ...useKibanaMock().services.application,
+      currentAppId$: of('securitySolutionUI'),
+      applications$: of(
+        new Map([
+          ['securitySolutionUI', { category: { label: 'Test' } } as unknown as PublicAppInfo],
+        ])
+      ),
+    };
+    useApplicationMock.mockReturnValue({ appId: 'testAppId' });
+    useCasesToastMock.mockReturnValue({
+      showInfoToast: mockedToastInfo,
+    });
+
+    wrapper = ({ children }) => {
```

Comment:
You can use the `createAppMockRenderer` helper tool. It will take care of everything for you. You can take a look at `x-pack/plugins/cases/public/common/use_cases_local_storage.test.tsx` to see how we use it.